File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FToolbar.java
package io.github.gofaith.faithdroid.FViews;

import android.graphics.drawable.Drawable;
import android.support.v7.widget.Toolbar;
import android.view.View;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.R;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.Toolkit;
import io.github.gofaith.faithdroid.UI.UIController;

public class FToolbar extends FView implements AttrSettable,AttrGettable {
    private Toolbar v;
    public FToolbar(UIController controller) {
        parentController =controller;
        v = (Toolbar) parentController.activity.getLayoutInflater().inflate(R.layout.my_toolbar, parentController.rootView, false);
        view=v;
        setElevation("4");
        parentController.activity.setSupportActionBar(v);
        parseSize("[-2,-1]");
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr){
                //----------------------------------------------
        }
        return "";
    }

    @Override
    public void setAttr(String attr, final String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr){
            // -------------------------------------------------------------------
            case "Title":
                parentController.activity.getSupportActionBar().setTitle(value);
                break;
            case "SubTitle":
                parentController.activity.getSupportActionBar().setSubtitle(value);
                break;
            case "Menu":
                parentController.optionMenus=value;
                break;
            case "BackNavigation":
                if (value.equals("true")) {
                    v.setNavigationOnClickListener(new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            parentController.activity.onBackPressed();
                        }
                    });
                }
                parentController.activity.getSupportActionBar().setDisplayHomeAsUpEnabled(value.equals("true"));
                break;
            case "NavigationIcon":
                Toolkit.file2Drawable(parentController, value, new Toolkit.OnDrawableReadyListener() {
                    @Override
                    public void onDrawableReady(Drawable drawable) {
                        v.setNavigationIcon(drawable);
                    }
                });
                break;
            case "NavigationOnClick":
                v.setNavigationOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Faithdroid.triggerEventHandler(value, "");
                    }
                });
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FToolbar.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.graphics.Color
import android.graphics.drawable.Drawable
import android.view.View
import androidx.appcompat.widget.Toolbar
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.R
import io.github.stevenzack.faithdroidapp.UI.*


class FToolbar(controller: UIController) : FView(), AttrSettable, AttrGettable {
    private val v: Toolbar
    private val navigationIconSrc: String? = null

    init {
        parentController = controller
        v = parentController!!.activity.getLayoutInflater().inflate(
            R.layout.my_toolbar,
            parentController!!.rootView,
            false
        ) as Toolbar
        view = v
        setElevation("4")
        parentController!!.activity.setSupportActionBar(v)
        parseSize("[-2,-1]")
    }

    override fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {
            //----------------------------------------------
            "Title" -> return v.getTitle().toString()
            "SubTitle" -> return v.getSubtitle().toString()
        }
        return ""
    }

    override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // -------------------------------------------------------------------
            "Title" -> parentController!!.activity.getSupportActionBar()!!.setTitle(value)
            "TitleColor" -> v.setTitleTextColor(Color.parseColor(value))
            "SubTitle" -> parentController!!.activity.getSupportActionBar()!!.setSubtitle(value)
            "SubTitleColor" -> v.setSubtitleTextColor(Color.parseColor(value))
            "Menu" -> parentController!!.optionMenus = value
            "BackNavigation" -> {
                if (value == "true") {
                    v.setNavigationOnClickListener(object : View.OnClickListener {
                        override fun onClick(v: View) {
                            parentController!!.activity.onBackPressed()
                        }
                    })
                }
                parentController!!.activity.getSupportActionBar()!!.setDisplayHomeAsUpEnabled(value == "true")
            }
            "NavigationIcon" -> {
                if (navigationIconSrc != null && value == navigationIconSrc) {
                    return
                }
                Toolkit.file2Drawable(parentController!!, value, {drawable->
                        v.setNavigationIcon(drawable)
                })
            }
            "NavigationOnClick" -> v.setNavigationOnClickListener(object : View.OnClickListener {
                override fun onClick(v: View) {
                    Faithdroid.triggerEventHandler(value, "")
                }
            })
        }
    }
}
